src/boot/bootctl.c

```diff
@@ -1924,11 +1974,23 @@ static int verb_install(int argc, char *argv[], void *userdata) {
 
         (void) sync_everything();
 
-        if (arg_touch_variables)
-                r = install_variables(arg_esp_path,
-                                      part, pstart, psize, uuid,
-                                      "/EFI/systemd/systemd-boot" EFI_MACHINE_TYPE_NAME ".efi",
-                                      install);
+        if (!arg_touch_variables)
+                return r;
+
+        if (streq_ptr(arg_arch, "all"))
```

Comment:
allowin an arch that is magic is evil... at the very least it should be  macro, so that it's obvious it's special. But actually I think there should be a separate boolean for the special cases. No need to shove all special cases into a single string.